tests/helpers/test_update_coordinator.py

```diff
@@ -224,3 +224,23 @@ async def test_refresh_recover(crd, caplog):
 
     assert crd.last_update_success is True
     assert "Fetching test data recovered" in caplog.text
+
+
+async def test_coordinator_entity(crd):
+    """Test the CoordinatorEntity class."""
+    entity = update_coordinator.CoordinatorEntity(crd)
+
+    assert entity.should_poll is False
+
+    crd.last_update_success = False
+    assert entity.available is False
+
+    await entity.async_update()
+    assert entity.available is True
+
+    with patch(
+        "homeassistant.helpers.entity.Entity.async_on_remove"
+    ) as mock_async_on_remove:
+        await entity.async_added_to_hass()
+
+    assert mock_async_on_remove.called
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6090606)